dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java

```diff
@@ -565,6 +802,36 @@ protected ColumnAndAlias getCoordinateColumn(QueryItem item, String suffix) {
     return ColumnAndAlias.EMPTY;
   }
 
+  @Override
+  protected String getColumnWithCte(QueryItem item, CteContext cteContext) {
+    List<String> columns = new ArrayList<>();
+
+    // Get the CTE definition for the item
+    CteDefinition cteDef = cteContext.getDefinitionByItemUid(computeKey(item));
+    if (cteDef == null) {
+      throw new IllegalArgumentException("CTE definition not found for item: " + item);
```

Comment:
How does it get propagated it this is null?
We should avoid HTTP 500 errors (if this is the case).
We have to ensure this returns 409 Conflict with e friendly message defined by and ErrorCode in `ErrorCode.java`